### DISCUSSION TITLE
Query on the dataset

### DISCUSSION THREAD

**Discussion by payelsantra** [safe-graph/GNN-FakeNews#15] [Q&A] [answered]
Hello, I recently went through the work User Preference-aware Fake Detection(UPFD). I was going through that git hub repository: https://github.com/safe-graph/GNN-FakeNews.git. I was trying to crawl the historical tweets using tweepy. I took news as a root node and tried to find the user-ids. I got the 1-hop connection of users and also found the follower network of those users. But cannot understand how to detect that, which followers are actually propagating the particular news (basically I am trying to find 2/3-hop connection). How to solve this problem?

**Comment by YingtongDou** [comment, marked as answer]
Hi, please read the Section 2.2 of [our paper](https://arxiv.org/pdf/2104.12259.pdf) for detailed info regarding building the news propagation path.

> Given a news piece on social media, the user exogenous context is composed of all users that engaged with the news. We utilize the retweet information of news pieces to build a news propagation graph. As the toy example of the news propagation graph shown in Figure 1, it is a tree-structured graph where the root node represents the news piece, and other nodes represent users who share the root news. In this paper, we investigate the fake news propagation on Twitter as a proof-of-concept use case. To build propagation networks in Twitter, we follow a similar strategy used in [8, 17, 26]. Specifically, we define a new piece as 𝑣1, and {𝑣2, . . . , 𝑣𝑛} as a list of users that retweeted 𝑣1 ordered by time. We define two following rules to determine the news propagation path:
>
> - For any account 𝑣𝑖, if 𝑣𝑖 retweets the same news later than at least one following accounts in {𝑣1, . . . , 𝑣𝑛}, we estimate the news spreads from the account with the latest timestamp to account 𝑣𝑖. Since the latest tweets are first presented in the timeline of the Twitter app, and thus have higher probabilities to be retweeted.
> - If account 𝑣𝑖 does not follow any accounts in the retweet sequences including the source account, we conservatively estimate the news spreads from the accounts with the most number of followers. Because tweets from accounts with more followers have a higher chance to be viewed/retweeted by other users according to the Twitter content distributing rules.
